plugins/modem-manager/fu-mm-device.h

```diff
@@ -11,27 +10,40 @@
 
 #include <libmm-glib.h>
 
-#define FU_MM_DEVICE_FLAG_DETACH_AT_FASTBOOT_HAS_NO_RESPONSE "detach-at-fastboot-has-no-response"
-#define FU_MM_DEVICE_FLAG_UNINHIBIT_MM_AFTER_FASTBOOT_REBOOT                                       \
-	"uninhibit-modemmanager-after-fastboot-reboot"
-#define FU_MM_DEVICE_FLAG_USE_BRANCH  "use-branch"
-#define FU_MM_DEVICE_FLAG_DISABLE_ZLP "disable-zlp"
-
 #define FU_TYPE_MM_DEVICE (fu_mm_device_get_type())
-G_DECLARE_FINAL_TYPE(FuMmDevice, fu_mm_device, FU, MM_DEVICE, FuDevice)
+G_DECLARE_DERIVABLE_TYPE(FuMmDevice, fu_mm_device, FU, MM_DEVICE, FuUdevDevice)
+
+#define FU_MM_DEVICE_FLAG_USE_BRANCH "use-branch"
+
+/* less ifdefs */
+#if !MM_CHECK_VERSION(1, 19, 1)
+#define MM_MODEM_FIRMWARE_UPDATE_METHOD_SAHARA (1 << 4)
+#endif
+#if !MM_CHECK_VERSION(1, 24, 0)
+#define MM_MODEM_FIRMWARE_UPDATE_METHOD_DFOTA	      (1 << 5)
+#define MM_MODEM_FIRMWARE_UPDATE_METHOD_CINTERION_FDL (1 << 6)
+#endif
```

Comment:
with the major overhaul anyway; how about we drop support for ancient versions and set a baseline to the new ones only?